src/learner-record/index.ts

```diff
@@ -19,6 +27,64 @@ export class LearnerRecord {
 		this._inviteFactory = inviteFactory
 	}
 
+	async patchModuleRecord(jsonPatch: JsonPatch[], moduleRecordId: number) {
+		try {
+			this.logger.debug(`Patching module record with ID ${moduleRecordId}`)
+			const res: ModuleRecord = await this._restService.patchWithJsonPatch(`/module_records/${moduleRecordId}`, jsonPatch)
+			return plainToInstance(ModuleRecord, res)
+		} catch (e) {
+			throw new Error(`An error occurred when trying to patch the module record: ${e}`)
+		}
+	}
+	
+	async createModuleRecord(moduleRecord: ModuleRecordInput): Promise<ModuleRecord> {
+		try {
+			this.logger.debug(`Creating module record for module ID ${moduleRecord.moduleId} and user ID ${moduleRecord.userId}`)
+			const res: ModuleRecord = await this.restService.post('/module_records', moduleRecord)
+			return plainToInstance(ModuleRecord, res)
+		} catch (e) {
+			throw new Error(`An error occurred when trying to create the module record: ${e}`)
+		}
+	}
+
+	async patchCourseRecord(jsonPatch: JsonPatch[], userId: string, courseId: string) {
+		try {
+			this.logger.debug(`Patching course record for course ID ${courseId} and user ID ${userId}`)
+			const res: CourseRecord = await this._restService.patchWithJsonPatch(`/course_records?courseId=${courseId}&userId=${userId}`, jsonPatch)
+			return plainToInstance(CourseRecord, res)
+		} catch (e) {
+			throw new Error(`An error occurred when trying to patch the course record: ${e}`)
+		}
+	}
+	
+	async createCourseRecord(courseRecord: CourseRecordInput) {
+		try {
+			this.logger.debug(`Creating course record for course ID ${courseRecord.courseId}`)
+			const res: CourseRecord = await this.restService.post('/course_records', courseRecord)
+			return plainToInstance(CourseRecord, res)
+		} catch (e) {
+			throw new Error(`An error occurred when trying to create the course record: ${e}`)
+		}
+	}
+	
+	async getCourseRecord(courseId: string, userId: string): Promise<CourseRecord|undefined> {
```

Comment:
Does this promise return `undefined` when there's no course by the Course ID (thus the `Promise<CourseRecord|undefined>`)?

Reply:
Yes, the `let courseRecord` at the top of the function defaults to `undefined`

Reply:
Ok cool.